sklearn/decomposition/dict_learning.py

```diff
@@ -415,9 +416,9 @@ def _update_dict(dictionary, Y, code, verbose=False, return_r2=False,
 
 
 def dict_learning(X, n_components, alpha, max_iter=100, tol=1e-8,
-                  method='lars', n_jobs=None, dict_init=None, code_init=None,
-                  callback=None, verbose=False, random_state=None,
-                  return_n_iter=False, positive_dict=False,
+                  method='lars', method_max_iter=1000, n_jobs=None,
+                  dict_init=None, code_init=None, callback=None, verbose=False,
+                  random_state=None, return_n_iter=False, positive_dict=False,
```

Comment:
You should add new parameter at the end of function signature to avoid breaking code

Reply:
fair, done!